src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,163 @@
+package duke;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+
+/**
+ * User Interface class, print beautiful output to user
+ */
+public class Ui {
+    private final String lines = "____________________________" +
+            "________________________________";
+    private final String endline = "___________________________" +
+            "_________________________________\n";
+
+    /**
+     * constructor
+     */
+    public Ui() {
+        System.out.println("Hello!!! I am Duke, a new born chatbot\n");
+        System.out.println("How may I serve you?");
+    }
+
+    /**
+     * Prints bye message
+     */
+    public void bye() {
+        System.out.println(lines);
+        System.out.println("Bye! See you again");
+        System.out.println(endline);
+    }
+
+    public void showAllTasks(TaskList taskList) {
+        System.out.println("Here are all your tasks:");
+        for (int i = 0; i < taskList.size(); i++) {
+            System.out.println((i + 1) + "." + taskList.get(i).toString());
+        }
+    }
+
+    /**
+     * Tells the user that the task is already done
+     * @param markTask current list of tasks
+     */
+    public void showMarkTaskAsAlreadyUndone(Task markTask) {
+        System.out.println("You have already done this task!");
+        System.out.println(markTask.markAsUndone());
+    }
+
+    /**
+     * Tells the user that the task is marked as done
+     * @param markTask current list of tasks
+     */
+    public void showMarkTaskAsUndone(Task markTask) {
+        System.out.println("Good Job! You have marked this task as done!");
+        System.out.println(markTask.markAsUndone());
+    }
+
+    /**
+     * Tells the user that the task is not done yet
+     * @param markTask current list of tasks
+     */
+    public void showMarkTaskAsAlreadyDone(Task markTask) {
```

Comment:
I like how you give very detailed method names to explain what the method does 👍

Reply:
Great JavaDocs done here 😄